src/ResultsPage.js

```diff
@@ -117,23 +128,49 @@ class ResultsPage extends Component {
         let breweryIbus = brewery.beers.map(beer => {
           return `${Math.round(beer.ibu / 10) * 10}-${(Math.round(beer.ibu / 10) * 10) + 9}`;
         });
-        if(breweryIbus.includes(this.state.ibuSelection)) {
+        if (breweryIbus.includes(this.state.ibuSelection)) {
           acc.push(brewery);
         }
         return acc;
-      }, [])
-      this.setState({filteredBreweries: filterResults}); //add filter by ABV to callback
+      }, []);
+      this.setState({ filteredBreweries: filterResults }, () => this.filterByStarred());
+    } else {
+      this.filterByStarred();
+    }
+  }
+
+  updateStarredList = (name, change) => {
+    // let list = JSON.parse(localStorage.getItem('userStarredBreweries')) || [];
+    let list = this.state.starredBreweries ? [...this.state.starredBreweries] : [];
+
+    console.log(`list: ${list}`);
+
+    if (change === 'add' && list.length > 0) {
+      list.push(name)
+    } else if (list.length > 0) {
```

Comment:
You shouldn't need to check the lengths of the array in either of the if/else conditions here. Try pulling those out and just checking `if (change === 'add')`

Reply:
Sounds great, thank you!